django/db/models/expressions.py

```diff
@@ -962,3 +986,5 @@ def asc(self):
 
     def desc(self):
         self.descending = True
+
+from django.db.models import fields
```

Comment:
This is very much work in progress.

I think we should move the lookup registration to some other place (django.db.models.__init__ perhaps) to avoid circular imports. The lookups <-> fields circularity is a bit nasty to have.